### PR TITLE
Fixed PBR implementation and added PBR test state

### PR DESCRIPTION
**Pull request by Ithyx**
The GLTF viewer PBR shader was quite broken for a few reasons, so I tried to rebuild a test state for PBR from scratch to see what exactly was failing. After correcting the shader, I decided it could be worth it to keep the new state in. Also, some misc. maintenance was made regarding dependencies and features (mainly RT stuff)

### CODE REVIEW

File: macha/src/pbr_test/shaders/pbr/pbr.frag [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    float NdotL = max(dot(data.N, data.L), 0.0);
+    Lo += (kd * data.albedo / M_PI + specular) * radiance * NdotL;
+
+    vec3 ambient = vec3(0.03) * data.albedo * data.ao;
+    vec3 color = ambient + Lo;
+
+    color = color / (color + vec3(1.0));
+
+    // vec3 color = phong(data);
+    // vec3 color = vec3(data.metallic, data.roughness, data.ao);
```
Comment (hvdbm) [inline macha/src/pbr_test/shaders/pbr/pbr.frag:148]: Commented code.

File: src/material.rs [hunk 2]
```diff
@@ -257,13 +256,13 @@ impl MaterialBuilder {
...
-            color_blend_attachment_state: *color_blend_attachment_state,
+            shader_stages: vec![vertex_shader_stage, fragment_shader_stage],
+            vertex_input_state_info,
+            input_assembly_state_info,
+            rasterizer_state_info,
+            multisampling_state_info,
+            depth_stencil_state_info,
+            color_blend_attachment_state,
             layout,
             cache: None, // @TODO(Ithyx): use pipeline cache plz
```
Comment (hvdbm) [inline src/material.rs:267]: plz

**Review by hvdbm — APPROVED** [review]

File: src/allocated_types.rs [hunk 3]
```diff
@@ -116,6 +116,7 @@ pub struct AllocatedBufferBuilder {
     pub name: String,
 }
 
+/// @TODO(Ithyx): create new type with MemoryLocation::GpuOnly
```
[the remark this answers is not in the document]
Comment (Ithyx) [reply src/allocated_types.rs:119]: Yes, but out of scope for this PR. Thanks for the reminder !

File: macha/src/pbr_test/shaders/pbr/pbr.frag (hunk 1 above)
Comment (Ithyx) [reply macha/src/pbr_test/shaders/pbr/pbr.frag]: Removed in 418fc0a182667207e181d577b46f47555b87e17d

File: src/material.rs (hunk 2 above)
Comment (Ithyx) [reply src/material.rs:267]: ok, but later :<

**Pull request merged**